truth_set/java/structure/organisation/complex-code/ECommerce.java

```diff
@@ -0,0 +1,224 @@
+}
+
+class OrderInput {
+    private Customer customer;
+    private List<OrderItem> items;
+    private Shipping shipping;
+    private List<Coupon> coupons;
+
+    public Customer getCustomer() { return customer; }
+    public void setCustomer(Customer c) { this.customer = c; }
+    public List<OrderItem> getItems() { return items; }
+    public void setItems(List<OrderItem> i) { this.items = i; }
+    public Shipping getShipping() { return shipping; }
+    public void setShipping(Shipping s) { this.shipping = s; }
+    public List<Coupon> getCoupons() { return coupons; }
+    public void setCoupons(List<Coupon> c) { this.coupons = c; }
+}
+
+public class ECommerce {
+    public Map<String, Object> processOrder(OrderInput order) {
+        Map<String, Object> finalOrder = new HashMap<>();
+        if (order == null || order.getCustomer() == null || order.getCustomer().getId() == null || 
+            order.getItems() == null || order.getItems().isEmpty()) {
+            return null;
+        }
+
+        finalOrder.put("customerId", order.getCustomer().getId());
+        finalOrder.put("customerName", order.getCustomer().getName() != null ? 
+                      order.getCustomer().getName() : "Valued Customer");
+
+        if (order.getCustomer().getLoyaltyStatus() != null && 
+            order.getCustomer().getLoyaltyStatus() != LoyaltyStatus.NONE) {
+            finalOrder.put("loyalty", "Status: " + order.getCustomer().getLoyaltyStatus());
+        }
+
+        Map<String, Object> items = new HashMap<>();
+        finalOrder.put("items", items);
+
+        for (int i = 0; i < order.getItems().size(); i++) {
+            OrderItem item = order.getItems().get(i);
+            if (item.getProductId() == null || item.getQuantity() <= 0) {
+                continue;
+            }
+
+            double itemTotal = 0;
+            Double productPrice = getProductPrice(item.getProductId());
+            if (productPrice == null) {
+                items.put("item_" + i, "Price Not Available");
+                continue;
+            }
+
+            itemTotal += productPrice * item.getQuantity();
+
+            if (item.getOptions() != null) {
+                for (Option option : item.getOptions()) {
+                    if (option.getOptionId() == null || option.getValue() == null) {
+                        continue;
+                    }
+                    Double optionPrice = getOptionPrice(option.getOptionId(), option.getValue());
+                    if (optionPrice != null) {
+                        itemTotal += optionPrice;
+                    }
+                }
+            }
+
+            if (item.getDiscounts() != null) {
+                for (Discount discount : item.getDiscounts()) {
+                    if (discount.getDiscountId() == null || discount.getAmount() <= 0) {
+                        continue;
+                    }
+                    itemTotal -= discount.getAmount();
+                }
+            }
+
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider breaking down complex method</b></div>
<div id="fix">

The `processOrder` method appears to be quite complex with multiple nested conditions and loops. Consider breaking it down into smaller, more focused methods for better maintainability and readability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -111,55 +111,24 @@
      public Map<String, Object> processOrder(OrderInput order) {
          Map<String, Object> finalOrder = new HashMap<>();
          if (!isValidOrder(order)) {
              return null;
          }
          processCustomerInfo(order, finalOrder);
          processOrderItems(order, finalOrder);
          processShipping(order, finalOrder);
          return finalOrder;
      }

      private boolean isValidOrder(OrderInput order) {
          return order != null && order.getCustomer() != null && 
                 order.getCustomer().getId() != null && 
                 order.getItems() != null && !order.getItems().isEmpty();
      }

      private void processCustomerInfo(OrderInput order, Map<String, Object> finalOrder) {
          finalOrder.put("customerId", order.getCustomer().getId());
          finalOrder.put("customerName", order.getCustomer().getName() != null ? 
                        order.getCustomer().getName() : "Valued Customer");
          if (order.getCustomer().getLoyaltyStatus() != null && 
              order.getCustomer().getLoyaltyStatus() != LoyaltyStatus.NONE) {
              finalOrder.put("loyalty", "Status: " + order.getCustomer().getLoyaltyStatus());
          }
      }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2527483385>#df23f9</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged